article_store/articles/models.py

```diff
@@ -0,0 +1,84 @@
+import re
+import uuid
+
+from django.core.validators import RegexValidator
+from django.db import models
+
+
+PREVIEW = 'preview'
+PUBLISHED = 'published'
+READY = 'ready'
+
+ARTICLE_STATUSES = (
+    (PREVIEW, 'preview'),
+    (PUBLISHED, 'published'),
+    (READY, 'ready'),
+)
+
+ARTICLE_ID_FORMAT = r'^[A-Za-z0-9\-._]+$'
```

Comment:
if this is not wide-enough, it can be expanded easily later e.g. adding `\+`, as long as it's URL-friendly

Reply:
I don't mind, I was going by [this](https://github.com/libero/walking-skeleton/blob/api/schema/articles.yaml#L34).